### ISSUE TITLE
BSRGAN article's hyperparams differ from JSON option files; which one is right?

### ISSUE THREAD

**Issue by umbertov** [cszn/KAIR#136]
Hi, I was checking training details on the [BSRGAN](https://arxiv.org/pdf/2103.14006.pdf) paper, and comparing them to `options/train_bsrgan_x4_gan.json`, I found a few differences which might be important for reproduction and research. I'm working with BSRGAN due to my master thesis project.

I refer to the end of section 5, at page 6, second column:
> * [...] we train the BSRGAN by minimizing a weighted combination of L1 loss, VGG perceptual loss and spectral norm-based least square **PatchGAN** loss [19] with weights 1, 1 and **0.1**, respectively. *

Here i notice two discrepancies:
* In the options file, `netD.net_type` is **`discriminator_unet`**, rather than the paper's **`discriminator_patchgan`**
* The GAN loss weight `D_lossfn_weight` is **1** instead of **0.1** as mentioned in the paper

Then, the next sentence is:
> *In particular, the VGG perceptual loss is operated on the fourth convolution before the **fourth** rather than the fifth maxpooling layer of the pre-trained 19-layer VGG model as it is more stable to prevent color shift issues*

But looking at the definition of `F` (the feature network used for perceptual loss), I would expect that `F_feature_layer` contains the index of a single layer. Instead I find 5 indices ` [2,7,16,25,34] `. 

My question is which one's correct? The info on the paper or the one of the JSON files?
It could have gone both ways: reporting correct values on the paper then uploading wrong/old version of the JSON file, or reporting old/wrong data on the paper while having uploaded the correct JSON file to this repo.

@cszn Since you're among the authors of the paper, could you shed some light on this?

**Comment by cszn** [comment]
There is no unique solution, the two settings can give rise to similar results.
